sol/private/sol_binary.bzl

```diff
@@ -119,6 +132,9 @@ def _run_solc(ctx):
         args.add("--combined-json")
         args.add_joined(ctx.attr.combined_json, join_with = ",")
 
+    if ctx.attr.optimize:
```

Comment:
I'm mirroring the `solc` behaviour because their flags are a bit unintuitive and Solidity devs have come to understand them.

Setting `--optimize-runs` without `--optimize` simply ignores the former. Another weird thing is that "runs" doesn't mean how many times the optimizer is run but how many times one expects to run the code `¯\_(ツ)_/¯`